### PR TITLE
#16 fix date not being displayed, type conversion

### PR DESCRIPTION
**Pull request by WitchDevelops**
* made sure that the names of variables match between the database and codebase
* convert Date to a locale string in a separate component

in the database it is stored as `date_of_birth`, type: Date, format: 'YYYY-MM-DD'

### CODE REVIEW

File: src/components/PetCard.tsx [hunk 1]
```diff
@@ -65,7 +67,7 @@ export const PetCard = ({
               )}
               <CardDescription className="flex items-center gap-1">
                 <TbClockHour3 />
-                {dateOfBirth}
+                <DateOfBirth dateOfBirth={date_of_birth} />
```
Comment (nikkehtine) [inline src/components/PetCard.tsx:70]: I would revert to how it was before since it's basically one line of code, so that it's easier to read

**Review by nikkehtine — CHANGES_REQUESTED** [review]

File: src/components/DateOfBirth.tsx
[the remark this answers is not in the document]
Comment (WitchDevelops) [reply src/components/DateOfBirth.tsx:1]: we will have more logic, the default view on the PetCard will be the calculated age, not date of birth: https://github.com/users/WitchDevelops/projects/2/views/1?pane=issue&itemId=67585410

[the remark this answers is not in the document]
Comment (jagodakubicka) [reply src/components/DateOfBirth.tsx:1]: I think we should keep it as separate Component, as we will need to add more logic to it soon

**Review by jagodakubicka — APPROVED** [review]

**Comment by WitchDevelops** [comment]
Overview of changes:

- a separate component for the date was a bad idea. I deleted it.
- instead I created two functions: one that takes the birth date and formats it (to: e.g. `December 12, 2010`), and the other one that calculates age and displays the result as `x years, y months`.
- both functions take input of the `Date` type (directly from the database) and return a type `string`.
- those functions are now in `utils/date`. I suggest we store reusable functions in the `utils` folder and a meaningful subfolder. This helps with modularization of the code and minimizes repeating code.

**How it looks now:**
_on PetCard:_
image
_on pet page:_
image

**Before changes:**
_on PetCard_ (empty):
image
_on pet page:_ (empty, only the date of birth):
image

**Review by nikkehtine — APPROVED** [review]

**Pull request merged**